elf/perf.go

```diff
@@ -45,6 +46,8 @@ struct event_sample {
 struct read_state {
 	void *buf;
 	int buf_len;
+	uint64_t data_head_initialized;
+	uint64_t data_head;
```

Comment:
Add a comment that this two fields are for the backward read: as opposed to normal ring buffers, backward read buffers don't update the read pointer when reading. So we keep the state externally here.

Reply:
done